server/src/main/java/com/thoughtworks/go/server/service/plugins/processor/elasticagent/v2/ElasticAgentProcessorConverterV2.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.service.plugins.processor.elasticagent.v2;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.thoughtworks.go.plugin.access.elastic.models.AgentMetadata;
+import com.thoughtworks.go.plugin.access.elastic.v2.AgentMetadataConverterV2;
+import com.thoughtworks.go.plugin.access.elastic.v2.AgentMetadataDTO;
+import com.thoughtworks.go.server.service.plugins.processor.elasticagent.ElasticAgentProcessorConverter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class ElasticAgentProcessorConverterV2 implements ElasticAgentProcessorConverter {
+    private static final Gson GSON = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().create();
+    private static final AgentMetadataConverterV2 agentMetadataConverter = new AgentMetadataConverterV2();
```

Comment:
`AgentMetadataDTO` and `AgentMetadataConverterV2` are defined by the EA extension V2. Reusing it here would mean going back to tying down elastic-agent extension with processor's contract. We should use a different one here, its alright if it looks exactly the same. Also, like I mentioned earlier. Since the processor's contract hasn't changed, we don't really need three versions of the processor, converters or this DTO.